server/src/main/java/org/elasticsearch/cluster/metadata/AutoExpandReplicas.java

```diff
@@ -110,20 +109,21 @@ public OptionalInt getDesiredNumberOfReplicas(IndexMetadata indexMetadata, Routi
                 numMatchingDataNodes++;
             }
         }
+        return calculateDesiredNumberOfReplicas(numMatchingDataNodes);
+    }
 
+    // Package private only for testing purposes!
+    int calculateDesiredNumberOfReplicas(int numMatchingDataNodes) {
```

Comment:
I was wondering if it is a good idea to test a private method/logic. I think having a small test like `testCalculateDesiredNumberOfReplica` justifies exposing this otherwise private logic, since the test makes it clear what is expected when calculating replicas. 

I went with package-private since the other alternative of using protected doesn't make much sense considering this is a record class.

Reply:
Yes, that is fine, we do that often.